qe_metrics/libs/database.py

```diff
@@ -1,36 +1,26 @@
-import os
 from types import TracebackType
-from typing import Optional, Type
+from typing import Any, Dict, Optional, Type
 
-
-from pony import orm
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker, scoped_session
 from pyaml_env import parse_config
 from simple_logger.logger import get_logger
+from urllib.parse import quote_plus
 
 from qe_metrics.utils.general import verify_config
-from qe_metrics.libs.database_mapping import DB_OBJECT
-
-
-LOGGER = get_logger(name="Database")
+from qe_metrics.libs.database_mapping import Base
 
 
 class Database:
     def __init__(self, config_file: str, verbose: bool) -> None:
-        """
-        Initialize the Database class
-
-        Args:
-            config_file (str): Path to the yaml file holding database and Jira configuration.
-            verbose (bool): Verbose output of database connection and transactions.
-        """
-        self.logger = get_logger(name=self.__class__.__module__)
-        orm.set_sql_debug(debug=verbose)
-        self.config_file = config_file
-        self.db_config = parse_config(path=self.config_file)["database"]
+        self.logger = get_logger(name=__name__)
+        self.connection_string = self.connection_string_builder(db_config=parse_config(path=config_file)["database"])
+        self.verbose = verbose
 
     def __enter__(self) -> "Database":
-        self.bind_local_db_connection() if self.db_config.get("local") else self.bind_remote_db_connection()
```

Comment:
`__enter__` and `__exit__` is now redundant, we should use `with session` from `sqlalchemy`.

```
def __init__(self, config_file: str, verbose: bool) -> None:
        self.logger = get_logger(name=__name__)
        self.connection_string = self.connection_string_builder(db_config=parse_config(path=config_file)["database"])
        self.verbose = verbose
        self.engine = create_engine(url=self.connection_string, echo=self.verbose)
        Base.metadata.create_all(bind=self.engine)

    def session(self):
        return Session(bind=self.engine)

```

and usage:

```
db = Database(config_file=config_file, verbose=verbose_db)
with db.session() as db_session, Jira(config_file=config_file) as jira:

```